packages/ember-metal/lib/property_get.js

```diff
@@ -96,6 +96,15 @@ export function get(obj, keyName) {
       if (DESCRIPTOR_TRAP && isDescriptorTrap(value)) {
         descriptor = value[DESCRIPTOR];
       } else if (isDescriptor(value)) {
+        deprecate(
+          `[DEPRECATED] computed property '${keyName}' was not set on object '${obj && obj.toString && obj.toString()}' via 'defineProperty'`,
+          !EMBER_METAL_ES5_GETTERS,
+          {
+            id: 'ember-meta.descriptor-on-object',
+            until: '3.5.0',
+            url: 'https://emberjs.com/deprecations/v3.x#toc_ember-computed-property-defineProperty'
```

Comment:
For the link to work correctly, this should be the downcased header of the deprecation. E.g.

> Use notifyPropertyChange instead of propertyWillChange and propertyDidChange

becomes

> #toc_use-notifypropertychange-instead-of-propertywillchange-and-propertydidchange

Reply:
@mmun thanks.  There will be an accompanying pr to emberjs/website although I need to work through exactly what the deprecation guide will be with @rwjblue.